### PR TITLE
Several fixes to make sure second updates work

### PR DESCRIPTION
**Pull request by kacf**
Ticket: MEN-6807

### CODE REVIEW

File: src/mender-update/http_resumer.hpp [hunk 1]
```diff
@@ -89,6 +89,7 @@ class DownloadResumerAsyncReader : virtual public io::AsyncReader {
 	// The header handler needs to manipulate inner_reader_ in order to replace it in
 	// subsequent requests.
```
Comment (lluiscampos) [inline src/mender-update/http_resumer.hpp:90]: Nit: please update the comment

File: src/mender-update/http_resumer/http_resumer.cpp [hunk 2]
```diff
@@ -374,7 +381,7 @@ error::Error DownloadResumerClient::AsyncCall(
 
 	if (!*cancelled_) {
 		return error::Error(
-			make_error_condition(errc::operation_in_progress), "HTTP call already ongoing");
+			make_error_condition(errc::operation_in_progress), "HTTP resumer call already ongoing");
```
Comment (lluiscampos) [inline src/mender-update/http_resumer/http_resumer.cpp:384]: Thank you

**Review by lluiscampos — APPROVED** [review]
The fixes look good.

What happened with the destructor? Why is not needed now to call `Cancel()`

**Comment by kacf** [comment]
> What happened with the destructor? Why is not needed now to call `Cancel()`

I discovered that it already calls `Cancel()`, just indirectly, through the inner reader which also gets destroyed. So the issue was that it wasn't destroyed to begin with.

**Comment by kacf** [comment]
Oh but hold on! That won't work if there is no inner reader, for example in between waits! I'll make a test case for that too.

**Comment by kacf** [comment]
Agreed to merge early to unblock, but I will follow up on the issue I posted above.

**Pull request merged**